core/src/main/java/io/github/thibaultbee/streampack/streamers/bases/BaseStreamer.kt

```diff
@@ -198,14 +192,25 @@ abstract class BaseStreamer(
     @RequiresPermission(Manifest.permission.RECORD_AUDIO)
     override fun configure(audioConfig: AudioConfig) {
         require(hasAudio) { "Do not need to set audio as it is a video only streamer" }
+        require(audioSource != null) { "Audio source must not be null" }
 
-        // Keep settings when we need to reconfigure
         this.audioConfig = audioConfig
 
         try {
-            audioSource?.configure(audioConfig)
-            audioEncoder?.release()
-            audioEncoder?.configure(audioConfig)
+            audioSource.configure(audioConfig)
+
+            this.audioEncoder?.release()
+            val audioEncoder =
+                MediaCodecEncoder(AudioEncoderConfig(audioConfig), listener = audioEncoderListener)
+            audioEncoder.configure()
+            (audioEncoder.input as MediaCodecEncoder.ByteBufferInput).listener =
```

Comment:
hmm, seems fishy to be having to do a cast here. why is input not a ByteBufferInput?

Reply:
Mostly because encoder is audio/video agnostic.
But it should returns an exception if input is not a `ByteBufferInput` here.